src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1270,6 +1286,22 @@ InferenceProfiler::ValidLatencyMeasurement(
         }
       }
     }
+    const auto& response_times = std::get<1>(timestamp);
+    // Gathering durations between responses. Need more than 1 element for
+    // metrics to be valid. The invariant for timestamps is that if the final
+    // response is a null response, it will NOT be added to the timestamp vector
+    if (response_times.size() > 1) {
+      for (size_t i = 0; i < response_times.size() - 1; i++) {
+        uint64_t curr_response_ns = CHRONO_TO_NANOS(response_times[i]);
+        uint64_t next_response_ns = CHRONO_TO_NANOS(response_times[i + 1]);
+        if (curr_response_ns <= next_response_ns) {
+          if ((curr_response_ns >= valid_range.first) &&
```

Comment:
Am I reading this correctly that responses that straddle windows will not be fully counted?
If the window is from 10 to 20, and responses are at 8,9,10,11,12, that the 8 and 9 will be dropped on the floor?

Reply:
As it is currently coded. 
How do you recommend we attribute the responses?

Reply:
well we definitely can't be dropping responses on the floor. That was an issue with the tool that we had to fix when we took over.

For coupled, if a req was in window A and the response was in window B, we would count the req and response in window B. I think we should continue to do that, but instead it should be `if a req was in window A and the FINAL response was in window B, we would count the req and all responses in window B`. We can't be dropping some of the initial responses that happened in window A and not count them toward either window.